### DISCUSSION TITLE
[Q] ERROR no such table: django_q_schedule

### DISCUSSION THREAD

**Discussion by amo13** [jonaswinkler/paperless-ng#1050] [Support] [answered]
Since I updated from 1.4.1 to 1.4.4, every 30 seconds `[Q] ERROR no such table: django_q_schedule` is logged by the scheduler.

Could this be a dependency issue? If so, any idea which one it could be? (I have the current django-q 1.3.6)

**Comment by TheFehr** [comment, marked as answer]
Missing tables normally suggests the database migrations not having worked properly. How do you have it installed?

If with docker-compose, try this command:
```docker-compose exec webserver python3 manage.py showmigrations```
This should give you a list of all migrationsland show if they were successfully run.

**Reply by amo13** [reply]
Thank you for your response.

I ran the migrations and no error occured. Here is the results of `showmigrations`
```
admin
 [X] 0001_initial
 [X] 0002_logentry_remove_auto_add
 [X] 0003_logentry_add_action_flag_choices
auth
 [X] 0001_initial
 [X] 0002_alter_permission_name_max_length
 [X] 0003_alter_user_email_max_length
 [X] 0004_alter_user_username_opts
 [X] 0005_alter_user_last_login_null
 [X] 0006_require_contenttypes_0002
 [X] 0007_alter_validators_add_error_messages
 [X] 0008_alter_user_username_max_length
 [X] 0009_alter_user_last_name_max_length
 [X] 0010_alter_group_name_max_length
 [X] 0011_update_proxy_permissions
 [X] 0012_alter_user_first_name_max_length
authtoken
 [X] 0001_initial
 [X] 0002_auto_20160226_1747
 [X] 0003_tokenproxy
contenttypes
 [X] 0001_initial
 [X] 0002_remove_content_type_name
django_q
 [X] 0001_initial
 [X] 0002_auto_20150630_1624
 [X] 0003_auto_20150708_1326
 [X] 0004_auto_20150710_1043
 [X] 0005_auto_20150718_1506
 [X] 0006_auto_20150805_1817
 [X] 0007_ormq
 [X] 0008_auto_20160224_1026
 [X] 0009_auto_20171009_0915
 [X] 0010_auto_20200610_0856
 [X] 0011_auto_20200628_1055
 [X] 0012_auto_20200702_1608
 [X] 0013_task_attempt_count
 [X] 0014_schedule_cluster
documents
 [X] 0001_initial
 [X] 0002_auto_20151226_1316
 [X] 0003_sender
 [X] 0004_auto_20160114_1844
 [X] 0005_auto_20160123_0313
 [X] 0006_auto_20160123_0430
 [X] 0007_auto_20160126_2114
 [X] 0008_document_file_type
 [X] 0009_auto_20160214_0040
 [X] 0010_log
 [X] 0011_auto_20160303_1929
 [X] 0012_auto_20160305_0040
 [X] 0013_auto_20160325_2111
 [X] 0014_document_checksum
 [X] 0015_add_insensitive_to_match
 [X] 0016_auto_20170325_1558
 [X] 0017_auto_20170512_0507
 [X] 0018_auto_20170715_1712
 [X] 0019_add_consumer_user
 [X] 0020_document_added
 [X] 0021_document_storage_type
 [X] 0022_auto_20181007_1420
 [X] 0023_document_current_filename
 [X] 1000_update_paperless_all
 [X] 1001_auto_20201109_1636
 [X] 1002_auto_20201111_1105
 [X] 1003_mime_types
 [X] 1004_sanity_check_schedule
 [X] 1005_checksums
 [X] 1006_auto_20201208_2209
 [X] 1007_savedview_savedviewfilterrule
 [X] 1008_auto_20201216_1736
 [X] 1009_auto_20201216_2005
 [X] 1010_auto_20210101_2159
 [X] 1011_auto_20210101_2340
 [X] 1012_fix_archive_files
 [X] 1013_migrate_tag_colour
 [X] 1014_auto_20210228_1614
 [X] 1015_remove_null_characters
 [X] 1016_auto_20210317_1351
paperless_mail
 [X] 0001_initial
 [X] 0002_auto_20201117_1334
 [X] 0003_auto_20201118_1940
 [X] 0004_mailrule_order
 [X] 0005_help_texts
 [X] 0006_auto_20210101_2340
 [X] 0007_auto_20210106_0138
 [X] 0008_auto_20210516_0940
sessions
 [X] 0001_initial
```

**Reply by TheFehr** [reply]
Well then I'm out of ideas. I have not yet had a deep dive into to code.

**Reply by amo13** [reply]
Also tried re-running the migrations and also running `makemigrations` but both commands have nothing do so `No migrations to apply` and `No changes detected`. 
Thank you anyway, maybe someone else has an idea.

**Reply by jonaswinkler** [reply]
All I know is that I delayed upgrades to django q because of some configuration issue / invalid warning. The version listed in the requirements is still 1.3.4, ~which does not have that new table.~

Make sure you're not mixing up multiple versions (such as in virtual environments). Running with 1.3.6 should be fine though.

**Reply by amo13** [reply]
Thank you for your reply.
I don't think I am mixing up versions since I only have one installed. I use the system packages (from distro repo) and have no virtual environment set up. Before the update, I was on Q 1.3.5 with paperless 1.4.1.
Maybe the message can be ignored, though it is flooding the logs

**Reply by amo13** [reply]
On another machine, I have paperless 1.4.4 (set up the same way) with Q 1.3.5 and I don't get the message about the missing table.

**Reply by jonaswinkler** [reply]
Scratch that about the table. The migration in django q 1.3.6 only added a field to the schedule table.

**Reply by amo13** [reply]
This is odd... I did a new installation of 1.4.3 in a VM, uploaded one document, updated to 1.4.5, and now I get the following message logged every 30s
```
[Q] ERROR no such column: django_q_schedule.cluster
```

On my productive instance, I still get `[Q] ERROR no such table: django_q_schedule` every 30s. 
But it looks like somehow my database got messed up, because I also have the [`BadSignature`](https://github.com/jonaswinkler/paperless-ng/issues/1110) error and the missing [`documents_document`](https://github.com/jonaswinkler/paperless-ng/issues/1106) error...
I'd really like to know what went wrong, so I can prevent it in the future, but it looks like it won't be easy to find out.

If I need to reinstall, is there a clean way to keep or reimport the database? I put quite some effort in setting up everything, tagging etc.

**Reply by amo13** [reply]
Ok, so the error does not show up if I rename `./paperless.conf` to `./paperless.conf.bak`. The update pulls the `paperless.conf` template in the paperless directory. 

What I did not expect is that this template `./paperless.conf` takes precedence before `/etc/paperless.conf`. So I got this bug because my paperless instance was using the template config instead of mine. 

So I must ask: shouldn't either the template `paperless.conf` file be renamed to `paperless.conf.template` or the precedence be changed to first `/etc/paperless.conf` and only second `./paperless.conf`?
